ssh-uuid.sh

```diff
@@ -184,20 +185,28 @@ function parse_args {
 		# Is arg a UUID.balena hostname specification?
 		# For ssh:
 		#   '[user@]UUID.balena'
+		#   '[user@]UUID.local'
 		#   'ssh://[user@]UUID.balena[:port]'
+		#   'ssh://[user@]UUID.local[:port]'
 		# For scp:
 		#   '[user@]UUID.balena:'
+		#   '[user@]UUID.local:'
 		#   'scp://[user@]UUID.balena[:port][/path]'
+		#   'scp://[user@]UUID.local[:port][/path]'
 		# where UUID is a hexadecimal number with exactly 32 or 62 characters,
-		# where 62 is not typo meant to read 64, it really is 62.
+		# where 62 is not typo meant to read 64, it really is 62. 7-character
+		# abbreviated UUIDs are also accepted for local-mode devices.
 		if [[ "${SSUU_SCP}" = 0 &&
-				"${arg}" =~ ^(ssh://)?((.+)@)?([[:xdigit:]]{32}|[[:xdigit:]]{62})\.balena(:[0-9]+)?$
+				"${arg}" =~ ^(ssh://)?((.+)@)?([[:xdigit:]]{7}|[[:xdigit:]]{32}|[[:xdigit:]]{62})\.(balena|local)(:[0-9]+)?$
```

Comment:
In a previous comment, I asked you to update the README file to clarify that 7-character UUIDs can be used with `.local` hostnames. However, hang on — I’ve thought of something. For a remote device reachable through the Balena cloud proxy server, the UUID has a special meaning as the identifier of the device in the cloud. Therefore, a regex ensuring a hexadecimal string of a certain length makes sense. However, for a `.local` hostname like `abc1234.local` (7-char UUID), the UUID is merely the device’s hostname discoverable through mDNS / Ahavi and configurable through the device’s [config.json](https://docs.balena.io/reference/OS/configuration/#hostname) file. In this case, `abc1234.local` (7-char hex) is just as good as `foo.local` (if `config.json` specified `"hostname": "foo"`), and a regex requiring a 7-char hex string would be unnecessarily restrictive.

Does that make sense? So the code could / should be changed to simply check for a `.local` hostname suffix, in which case any string should be allowed as the hostname prefix. Then the README could be updated along the following lines:

Before:
> The device's ssh/scp hostname has the format `'<device-UUID>.balena'`, using the device's
-full UUID (not a shortened form).

After:
> The device’s ssh/scp hostname should have one of the following formats:
> - `'<device-UUID>.balena'` - Remote devices reachable through Balena’s cloud proxy server.
> - `'<hostname>.local'` - Devices on the same local network as the client.
>
> For remote devices, `<device-UUID>` must be the device’s “full” UUID (32 or 62 characters long). For local devices, `<hostname>` is the device’s hostname configured in its [config.json](https://docs.balena.io/reference/OS/configuration/#hostname) file — by default, it is the device’s 7-character short UUID.

Reply:
Also, in this PR’s initial approach, `abc1234.balena` would not fail the regex check, but it should fail the regex check in order to alert the user of a hostname problem.

Reply:
Sure thing! That sounds like a very reasonable approach.